src/components/code/code_block.tsx

```diff
@@ -402,57 +412,62 @@ const useFullScreen = ({
 
   const showFullScreenButton = !!overflowHeight;
 
-  const FullScreenButton: React.FC = () => {
-    if (!showFullScreenButton) return null;
-    return (
-      <EuiI18n
-        tokens={[
-          'euiCodeBlock.fullscreenCollapse',
-          'euiCodeBlock.fullscreenExpand',
-        ]}
-        defaults={['Collapse', 'Expand']}
-      >
-        {([fullscreenCollapse, fullscreenExpand]: string[]) => (
-          <EuiButtonIcon
-            className="euiCodeBlock__fullScreenButton"
-            onClick={toggleFullScreen}
-            iconType={isFullScreen ? 'fullScreenExit' : 'fullScreen'}
-            color="text"
-            aria-label={isFullScreen ? fullscreenCollapse : fullscreenExpand}
-          />
-        )}
-      </EuiI18n>
-    );
-  };
-
-  const FullScreenDisplay: React.FC<{ className: string }> = ({
-    children,
-    className,
-  }) => {
-    if (!isFullScreen) return null;
-
-    // Force fullscreen to use large font and padding.
-    const fullScreenClasses = classNames(
-      className,
-      'euiCodeBlock--fontLarge',
-      'euiCodeBlock--paddingLarge',
-      'euiCodeBlock-isFullScreen'
-    );
-
-    // Attaches to the body because of EuiOverlayMask's React portal usage.
-    return (
-      <EuiOverlayMask>
-        <EuiFocusTrap clickOutsideDisables={true}>
-          <div className={fullScreenClasses}>{children}</div>
-        </EuiFocusTrap>
-      </EuiOverlayMask>
-    );
-  };
+  const _FullScreenButton = useMemo(() => {
+    const FullScreenButton: FunctionComponent = () => {
```

Comment:
Same q here about `useCallback`
```suggestion
  const FullScreenButton = useCallback(() => {
```